diff --git a/lnclient/lnd/lnd.go b/lnclient/lnd/lnd.go
index bad532f..ba35b7b 100644
--- a/lnclient/lnd/lnd.go
+++ b/lnclient/lnd/lnd.go
@@ -708,7 +708,7 @@ func (svc *LNDService) MakeInvoice(ctx context.Context, amountMsat int64, descri
 		DescriptionHash: descriptionHashBytes,
 		Expiry:          expiry,
 		RouteHints:      hints,
-		Private:         !hasPublicChannels, // use private channel hints in the invoice
+		Private:         true, // always include private channel hints in the invoice
 	}
 
 	resp, err := svc.client.AddInvoice(ctx, addInvoiceRequest)
