diff --git a/cmd/http/main.go b/cmd/http/main.go
index ed31f96..4d31212 100644
--- a/cmd/http/main.go
+++ b/cmd/http/main.go
@@ -2,8 +2,8 @@ package main
 
 import (
 	"context"
-	"fmt"
 	nethttp "net/http"
+	"net"
 	"os"
 	"os/signal"
 	"syscall"
@@ -51,8 +51,9 @@ func main() {
 	httpSvc := http.NewHttpService(svc, svc.GetEventPublisher())
 	httpSvc.RegisterSharedRoutes(e)
 	//start Echo server
+	bindAddress := net.JoinHostPort(svc.GetConfig().GetEnv().Host, svc.GetConfig().GetEnv().Port)
 	go func() {
-		if err := e.Start(fmt.Sprintf(":%v", svc.GetConfig().GetEnv().Port)); err != nil && err != nethttp.ErrServerClosed {
+		if err := e.Start(bindAddress); err != nil && err != nethttp.ErrServerClosed {
 			logger.Logger.WithError(err).Error("echo server failed to start")
 			cancel()
 		}
diff --git a/config/models.go b/config/models.go
index 0fb8870..dc11afd 100644
--- a/config/models.go
+++ b/config/models.go
@@ -24,6 +24,7 @@ type AppConfig struct {
 	LNDAddress                         string `envconfig:"LND_ADDRESS"`
 	LNDCertFile                        string `envconfig:"LND_CERT_FILE"`
 	LNDMacaroonFile                    string `envconfig:"LND_MACAROON_FILE"`
+	Host                               string `envconfig:"HOST" default:"127.0.0.1"`
 	Workdir                            string `envconfig:"WORK_DIR"`
 	Port                               string `envconfig:"PORT" default:"8080"`
 	DatabaseUri                        string `envconfig:"DATABASE_URI" default:"nwc.db"`
