diff --git a/frontend/frontend.go b/frontend/frontend.go
index 203dd9b..9c7f6da 100644
--- a/frontend/frontend.go
+++ b/frontend/frontend.go
@@ -1,26 +1,7 @@
 package frontend
 
 import (
-	"embed"
-	"net/http"
-
 	"github.com/labstack/echo/v4"
-	"github.com/labstack/echo/v4/middleware"
 )
 
-//go:embed dist
-var embeddedReactAssets embed.FS
-
-func RegisterHandlers(e *echo.Echo) {
-	e.Use(middleware.StaticWithConfig(middleware.StaticConfig{
-		Skipper: nil,
-		// Root directory from where the static content is served.
-		Root: "dist",
-		// Enable HTML5 mode by forwarding all not-found requests to root so that
-		// SPA (single-page application) can handle the routing.
-		HTML5:      true,
-		Browse:     false,
-		IgnoreBase: false,
-		Filesystem: http.FS(embeddedReactAssets),
-	}))
-}
+func RegisterHandlers(e *echo.Echo) {}
diff --git a/http/http_service.go b/http/http_service.go
index 63d0ffc..0621bf5 100644
--- a/http/http_service.go
+++ b/http/http_service.go
@@ -114,8 +114,6 @@ func (httpSvc *HttpService) RegisterSharedRoutes(e *echo.Echo) {
 	e.POST("/api/backup", httpSvc.createBackupHandler, unlockRateLimiter)
 	e.GET("/logout", httpSvc.logoutHandler)
 
-	frontend.RegisterHandlers(e)
-
 	// restricted routes
 	// Configure middleware with the custom claims type
 	jwtConfig := echojwt.Config{
